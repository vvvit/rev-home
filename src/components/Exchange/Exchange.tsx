import * as React from 'react';
import {cn} from '@bem-react/classname';

import { CurrencyCard, CurrencyCardType } from '../CurrencyCard/CurrencyCard';

const cnApp = cn('Exchange');

class Exchange extends React.Component {
  render() {
    return (
      <div className={cnApp()}>
        <CurrencyCard
            currency="RUR"
            type={CurrencyCardType.buy}
        />
        <CurrencyCard
            currency="USD"
            type={CurrencyCardType.sell}
        />
      </div>
    );
  }
}

export { Exchange };
