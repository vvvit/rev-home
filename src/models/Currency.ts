export const enum CurrencyName {
    USD = 'usd',
    EUR = 'eur',
    GBP = 'gbp',
}

export interface ICurrency {
    name: CurrencyName,
    value: number
}
