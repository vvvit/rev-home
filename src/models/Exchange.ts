export const enum CurrencyExchangeType {
    BUY = 'buy',
    SELL = 'sell'
}
