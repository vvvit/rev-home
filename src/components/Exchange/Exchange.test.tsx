import * as React from 'react';
import { shallow } from 'enzyme';

import { Exchange } from './Exchange';
import { CurrencyName } from '../../models/Currency';

describe('Exchange', () => {
    let updateRates: jest.Mock;

    beforeEach(() => {
        updateRates = jest.fn();
    });

    it('renders correctly', () => {
        expect(shallow(
            <Exchange
                buyCurrency={CurrencyName.EUR}
                sellCurrency={CurrencyName.USD}
                updateRates={updateRates}
            />
        )).toMatchSnapshot();
    });

    it('updates rate when buy currency changed', () => {
        const component = shallow(
            <Exchange
                buyCurrency={CurrencyName.EUR}
                sellCurrency={CurrencyName.USD}
                updateRates={updateRates}
            />
        );

        component.setProps({ buyCurrency: CurrencyName.USD });

        expect(updateRates).toBeCalled();
    });

    it('updates rate when sell currency changed', () => {
        const component = shallow(
            <Exchange
                buyCurrency={CurrencyName.EUR}
                sellCurrency={CurrencyName.USD}
                updateRates={updateRates}
            />
        );

        component.setProps({ sellCurrency: CurrencyName.EUR });

        expect(updateRates).toBeCalled();
    });
});
