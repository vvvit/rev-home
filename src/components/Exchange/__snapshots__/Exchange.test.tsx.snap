// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exchange renders correctly 1`] = `
<div
  className="Exchange"
>
  <Connect(CurrencyCard)
    type="sell"
  />
  <Connect(Rate) />
  <Connect(CurrencyCard)
    type="buy"
  />
</div>
`;
